streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -728,33 +745,57 @@ int maybeCommitActiveTasksPerUserRequested() {
             return -1;
         } else {
-            final Map<TaskId, Map<TopicPartition, OffsetAndMetadata>> consumedOffsetsAndMetadataPerTask = new HashMap<>();
-            for (final Task task : activeTaskIterable()) {
-                if (task.commitRequested() && task.commitNeeded()) {
-                    task.prepareCommit();
-                    final Map<TopicPartition, OffsetAndMetadata> offsetAndMetadata = task.committableOffsetsAndMetadata();
-                    if (!offsetAndMetadata.isEmpty()) {
-                        consumedOffsetsAndMetadataPerTask.put(task.id(), offsetAndMetadata);
+            if (eosBetaEnabled && !eosUpgradeModeEnabled) {
+                for (final Task task : activeTaskIterable()) {
+                    if (task.commitRequested() && task.commitNeeded()) {
+                        return commitAll();
                     }
                 }
-            }
 
-            commitOffsetsOrTransaction(consumedOffsetsAndMetadataPerTask);
+                return 0;
+            } else {
+                final Map<TaskId, Map<TopicPartition, OffsetAndMetadata>> consumedOffsetsAndMetadataPerTask = new HashMap<>();
+
+                for (final Task task : activeTaskIterable()) {
+                    if (task.commitRequested() && task.commitNeeded()) {
+                        task.prepareCommit();
+                        final Map<TopicPartition, OffsetAndMetadata> offsetAndMetadata = task.committableOffsetsAndMetadata();
+                        if (!offsetAndMetadata.isEmpty()) {
+                            consumedOffsetsAndMetadataPerTask.put(task.id(), offsetAndMetadata);
+                        }
+                    }
+                }
 
-            for (final Task task : tasks.values()) {
-                if (consumedOffsetsAndMetadataPerTask.containsKey(task.id())) {
-                    task.postCommit();
+                commitOffsetsOrTransaction(consumedOffsetsAndMetadataPerTask);
+
+                for (final Task task : tasks.values()) {
+                    if (consumedOffsetsAndMetadataPerTask.containsKey(task.id())) {
+                        task.postCommit();
+                    }
                 }
-            }
 
-            return consumedOffsetsAndMetadataPerTask.size();
+                return consumedOffsetsAndMetadataPerTask.size();
+            }
         }
     }
 
     private void commitOffsetsOrTransaction(final Map<TaskId, Map<TopicPartition, OffsetAndMetadata>> offsetsPerTask) {
-        if (eosEnabled) {
+        if (eosAlphaEnabled) {
             for (final Map.Entry<TaskId, Map<TopicPartition, OffsetAndMetadata>> taskToCommit : offsetsPerTask.entrySet()) {
-                activeTaskCreator.streamsProducerForTask(taskToCommit.getKey()).commitTransaction(taskToCommit.getValue());
+                activeTaskCreator.streamsProducerForTask(taskToCommit.getKey())
+                    .commitTransaction(taskToCommit.getValue(), mainConsumer.groupMetadata().groupId());
```

Comment:
Eos-alpha: we just pass `application.id` during per-task tx-commit

Reply:
See my other comment: I think in 2.6+ we would always pass consumer metadata no matter of the eos mode.

And a meta comment here: we should avoid unnecessary branches if their effects are minimal, to keep the logic simple and less error-prone.